lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/FunctionalChaosResult.cxx

```diff
@@ -307,6 +313,130 @@ void FunctionalChaosResult::setInvolvesModelSelection(const Bool involvesModelSe
+{
+  // Get marginal input sample and distribution
+  const UnsignedInteger inputDimension = inputSample_.getDimension();
+  conditioningIndices.check(inputDimension);
+  const Sample inputSampleMarginal(inputSample_.getMarginal(conditioningIndices));
+  const Distribution inputDistributionMarginal(distribution_.getMarginal(conditioningIndices));
+
+  // Restrict the orthogonal basis
+  // TODO: create a new OrthogonalBasis.getMarginal(activeIndices)
+  const String basicClassName(orthogonalBasis_.getImplementation()->getClassName());
+  if (basicClassName != "OrthogonalProductPolynomialFactory")
+    throw InvalidArgumentException(HERE) << "This class can only manage an OrthogonalProductPolynomialFactory "
+          << "but current basis is" << basicClassName;
+  const OrthogonalProductPolynomialFactory* p_basis = dynamic_cast<const OrthogonalProductPolynomialFactory*>(orthogonalBasis_.getImplementation().get());
+  const OrthogonalProductPolynomialFactory::PolynomialFamilyCollection polynomialCollection(p_basis->getPolynomialFamilyCollection());
+
+  // Restrict the collection of polynomial families
+  OrthogonalProductPolynomialFactory::PolynomialFamilyCollection polynomialMarginalCollection;
+  for (UnsignedInteger index = 0; index < inputDimension; ++ index)
+    if (conditioningIndices.contains(index))
+      polynomialMarginalCollection.add(polynomialCollection[index]);
+
+  // Restrict the enumeration function
+  // TODO: create a new EnumerateFunction.getMarginal(activeIndices)
+  const EnumerateFunction enumerateFunction(orthogonalBasis_.getEnumerateFunction());
+  const EnumerateFunctionImplementation enumerateFunctionImplementation(enumerateFunction.getImplementation());
+  const String enumerateName(enumerateFunction.getImplementation()->getClassName());
+  EnumerateFunction enumerateFunctionMarginal;
+  const UnsignedInteger activeDimension = conditioningIndices.getSize();
+  if (enumerateName == "LinearEnumerateFunction")
+  {
+    enumerateFunctionMarginal = LinearEnumerateFunction(activeDimension);
+  }
+  else if (enumerateName == "HyperbolicAnisotropicEnumerateFunction")
+  {
+    const HyperbolicAnisotropicEnumerateFunction* p_rule = dynamic_cast<const HyperbolicAnisotropicEnumerateFunction*>(enumerateFunction.getImplementation().get());
+    const Scalar quasiNorm(p_rule->getQ());
+    const Point weightVector(p_rule->getWeight());
+    Point weightMarginal(activeDimension);
+    for (UnsignedInteger i = 0; i < activeDimension; ++i)
+      weightMarginal[i] = weightVector[conditioningIndices[i]];
+    enumerateFunctionMarginal = HyperbolicAnisotropicEnumerateFunction(weightMarginal, quasiNorm);
+  }
+  else if (enumerateName == "NormInfEnumerateFunction")
+  {
+    enumerateFunctionMarginal = NormInfEnumerateFunction(activeDimension);
+  }
+  else
+    throw InvalidArgumentException(HERE) << "Cannot manage an enumerate function with "
+      << enumerateName;
```

Comment:
looks like this could be simplified now

Reply:
Fixed at 16078c089933ce670381d73b7d1dd5eff294376d